test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -109,9 +112,9 @@ func ClusterUpgradeWithRuntimeSDKSpec(ctx context.Context, inputGetter func() Cl
 	)
 
 	var (
-		input         ClusterUpgradeWithRuntimeSDKSpecInput
-		namespace     *corev1.Namespace
-		cancelWatches context.CancelFunc
+		input                             ClusterUpgradeWithRuntimeSDKSpecInput
+		namespace, infraNamespace         *corev1.Namespace
```

Comment:
Maybe let's also call this clusterClassNamespace
(and cancelClusterClassNamespaceWatches below)